onnxruntime/python/tools/transformers/fusion_skiplayernorm.py

```diff
@@ -56,19 +56,6 @@ def fuse(self, node, input_name_to_nodes, output_name_to_node):
         # Root Mean Square Layer Normalization
         simplified = node.op_type == "SimplifiedLayerNormalization"
 
-        if self.shape_infer_helper is not None:
```

Comment:
The shape check is still required. I think currently we support broadcast in kernel, so this logic can be slightly. How about change add a parameter allow_broadcast to compare_shape?

Currently SkipLayerNorm only allows 3D inputs, while LayerNorm allows other dimension.
We might also need add a check of dimensions like 
```
if self.shape_infer_helper.get_edge_shape(add.input[0]) and len(self.shape_infer_helper.get_edge_shape(add.input[0])) != 3:
    return
```

Reply:
[This PR](https://github.com/microsoft/onnxruntime/pull/16510) says broadcasting support for the above shapes has been added for CPU and CUDA.

Reply:
[This PR](https://github.com/microsoft/onnxruntime/pull/16510) says broadcasting support for the above shapes has been added for CPU and CUDA. I've added a workaround to keep the shape check.